identity/handler_test.go

```diff
@@ -1492,51 +1488,42 @@ func TestHandler(t *testing.T) {
 				t.Parallel()
 				body, res := getFull(t, ts, fmt.Sprintf("/identities?per_page=%d", perPage), http.StatusOK)
 				assert.Len(t, body.Array(), perPage)
-				assert.Equal(t, strconv.Itoa(count), res.Header.Get("X-Total-Count"))
+				_ = res
 			})
 		}
 
 		t.Run("iterate over next page", func(t *testing.T) {
 			perPage := 10
-			pagePath := fmt.Sprintf("/identities?per_page=%d", perPage)
 
-			run := func(t *testing.T, path string, knownIDs map[string]struct{}) (isLast bool, parsed *url.URL) {
-				var err error
+			run := func(t *testing.T, path string, knownIDs map[string]struct{}) (next *url.URL, res *http.Response) {
 				t.Logf("Requesting %s", path)
 				body, res := getFull(t, ts, path, http.StatusOK)
-				for _, link := range linkheader.Parse(res.Header.Get("Link")) {
-					if link.Rel != "next" {
-						isLast = true
-						continue
-					}
-					parsed, err = url.Parse(link.URL)
-					require.NoError(t, err)
-					isLast = false
-					break
-				}
-
 				for _, i := range body.Array() {
-					assert.NotContains(t, knownIDs, i.Get("id").String())
-					knownIDs[i.Get("id").String()] = struct{}{}
+					id := i.Get("id").String()
+					_, seen := knownIDs[id]
+					require.Falsef(t, seen, "ID %s was previously returned from the API", id)
+					knownIDs[id] = struct{}{}
+				}
+				links := link.ParseResponse(res)
+				if link, ok := links["next"]; ok {
+					next, err := url.Parse(link.URI)
+					require.NoError(t, err)
+					return next, res
 				}
-				return isLast, parsed
+				return nil, res
 			}
 
 			t.Run("using token pagination", func(t *testing.T) {
 				knownIDs := make(map[string]struct{})
-				var isLast bool
 				var pages int
-				path := pagePath
-				for !isLast {
-					t.Run(fmt.Sprintf("page=%d", pages), func(t *testing.T) {
-						var res *url.URL
-						pages++
-						isLast, res = run(t, path, knownIDs)
-						if isLast {
-							return
-						}
-						path = fmt.Sprintf("/identities?page_size=%s&page_token=%s", res.Query().Get("page_size"), res.Query().Get("page_token"))
-					})
+				path := fmt.Sprintf("/identities?page_size=%d", perPage)
+				for {
+					pages++
+					next, _ := run(t, path, knownIDs)
```

Comment:
Add explicit assertion to ensure page and per_page are not set, or we might still use page pagination here.

Reply:
Fixed.